src/ui/display.c

```diff
@@ -330,9 +354,34 @@ bool prepare_display_output() {
         strlcpy(symbol, token->symbol, MAX_TOKEN_SYMBOL_LEN + 1);
         symbol_len = strlen(token->symbol);
     }
-    strlcpy(g_amount, symbol, MAX_TOKEN_SYMBOL_LEN + 1);
-    g_amount[symbol_len] = ' ';
-    format_value(output.value, g_amount + symbol_len + 1);
+
+    if (is_authority_output(output.token_data)) {
+        g_is_authority = true;
+        // set g_authority
+        strlcpy(g_authority, symbol, MAX_TOKEN_SYMBOL_LEN + 1);
+        g_authority[symbol_len] = ' ';
+        if (is_mint_authority(output.token_data, output.value)) {
+            strlcpy(g_authority + symbol_len + 1, "Mint", 30);
+        } else {
+            if (is_melt_authority(output.token_data, output.value)) {
+                strlcpy(g_authority + symbol_len + 1, "Melt", 30);
+            } else {
+                // This authority is unknown, so we treat it as invalid
+                PRINTF("[-] Unknown authority received in value %d\n", output.value);
+                explicit_bzero(&G_context, sizeof(G_context));
```

Comment:
This is clearing the global context.
The `G_token_symbols` can be cleared but it is not a security issue to leave them initialized since only user confirmed tokens enter the `G_token_symbols` context.